File: adapter/dart/adapter.dart
main() {
  final specialLibrary = SpecialLibrary();
  final adapter = Adapter(specialLibrary);

  print(adapter.getData());
}

class SpecialLibrary {
  String getData() => "atad litu";
}

abstract class Data {
  String getData();
}

class Adapter implements Data {
  final SpecialLibrary library;

  Adapter(this.library);

  String getData() {
    final rawData = library.getData();
    return rawData.splitMapJoin(RegExp('i'),
        onMatch: (m) => '${m[0]}', onNonMatch: (n) => '*');
  }
}

File: adapter/kotlin/Adapter.kt
fun main(){
    val specialLibrary = SpecialLibrary()
    val adapter = Adapter(specialLibrary)

    print(adapter.getData())

}

class SpecialLibrary(){
    fun getData(): String = "atad litu"
}

interface Data{
    fun getData(): String
}


class Adapter(private val library: SpecialLibrary) : Data{

    override fun getData(): String{
        val rawData = library.getData()
        return rawData.reversed()
    }

}